### PR TITLE
multi: add a Status server

### LINKED ISSUE #290: litd exits with code 0 when it stops due to non-0 lnd exit
litd stops with a "successful" exit code 0 when it stops upon lnd exiting due to a failure and a non-0 exit status code.

**Expected behavior**

litd to exit with a non-0 exit code since it's stopping unexpectedly (and lnd presumably stopped unexpectedly as well).

**Actual behavior**

litd exits with a 0 exit code.

**To reproduce**

Have litd and lnd running - I have them running in systemd. Shutting down bitcoind causes lnd to exit with a failure status code while litd exits with a 0 code.

**System information**

I'm using litd binaries downloaded from github with a separate standalone lnd instance. This is on a rpi4 arm64.

Below are systemd status logs from lnd and litd (with some extra log output for litd) that shows lnd having exited with `1/FAILURE` while litd exits with `0/SUCCESS`.

```
$ sudo systemctl status lnd
● lnd.service - LND Lightning Daemon
   Loaded: loaded (/etc/systemd/system/lnd.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Mon 2021-12-06 10:07:32 EST; 55s ago
  Process: 21935 ExecStart=/home/bitcoin/go/bin/lnd (code=exited, status=1/FAILURE)
 Main PID: 21935 (code=exited, status=1/FAILURE)
```

```
$ sudo systemctl status lit
● lit.service - LND Lightning Terminal
   Loaded: loaded (/etc/systemd/system/lit.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Mon 2021-12-06 10:06:51 EST; 3min 24s ago
  Process: 15969 ExecStart=/home/bitcoin/lit/litd (code=exited, status=0/SUCCESS)
 Main PID: 15969 (code=exited, status=0/SUCCESS)

2021-12-06 10:06:48.679 [INF] SGNL: Received terminated
2021-12-06 10:06:48.820 [INF] SGNL: Shutting down...
2021-12-06 10:06:48.856 [INF] SGNL: Gracefully shutting down.
2021-12-06 10:06:48.857 [INF] LITD: Shutdown signal received
2021-12-06 10:06:48.883 [INF] LOOPD: Stopping gRPC server
2021-12-06 10:06:48.883 [INF] LOOPD: Stopping REST server
2021-12-06 10:06:48.883 [INF] LOOPD: Liquidity manager stopped
2021-12-06 10:06:48.885 [INF] LOOP: Swap client terminating
2021-12-06 10:06:48.885 [INF] LOOP: Swap client terminated
2021-12-06 10:06:48.885 [INF] LOOPD: Swap client stopped
2021-12-06 10:06:48.927 [ERR] ACCT: Unable to receive block notification: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:39338->127.0.0.1:10009: use of closed network connection
2021-12-06 10:06:48.884 [ERR] RPCS: Unable to receive block notification: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:39338->127.0.0.1:10009: use of closed network connection
2021-12-06 10:06:48.979 [ERR] RPCS: Unable to receive block notification: channel acceptor receive failed: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:39338->127.0.0.1:10009: use of closed network connection
2021-12-06 10:06:48.979 [INF] LOOPD: Daemon exited
2021-12-06 10:06:48.979 [INF] POOL: Received shutdown signal, stopping server
2021-12-06 10:06:48.979 [INF] AUCT: Shutting down auctioneer client
2021-12-06 10:06:48.979 [INF] RPCS: Trader server stopping
Dec 06 10:06:48 satoshpi litd[15969]: 2021-12-06 10:06:48.979 [INF] AUCT: Shutting down auctioneer client
Dec 06 10:06:48 satoshpi litd[15969]: 2021-12-06 10:06:48.980 [INF] FNDG: Funding manager stopping
Dec 06 10:06:48 satoshpi litd[15969]: 2021-12-06 10:06:48.980 [INF] FNDG: Stopped funding manager
Dec 06 10:06:48 satoshpi litd[15969]: 2021-12-06 10:06:48.981 [INF] RPCS: Stopped trader server
Dec 06 10:06:48 satoshpi litd[15969]: 2021-12-06 10:06:48.981 [ERR] LNDC: Error closing lnd connection: rpc error: code = Canceled desc = grpc: the client connection is closing
Dec 06 10:06:51 satoshpi litd[15969]: 2021-12-06 10:06:51.067 [ERR] LNDC: Error closing lnd connection: rpc error: code = Canceled desc = grpc: the client connection is closing
Dec 06 10:06:51 satoshpi systemd[1]: lit.service: Succeeded.
Dec 06 10:06:51 satoshpi systemd[1]: Stopped LND Lightning Terminal.
Dec 06 10:07:32 satoshpi systemd[1]: Dependency failed for LND Lightning Terminal.
Dec 06 10:07:32 satoshpi systemd[1]: lit.service: Job lit.service/start failed with result 'dependency'.
```

### LINKED ISSUE #420: Show LND/LiT version mismatch error in Web UI
**Is your feature request related to a problem? Please describe.**
Yes, I installed LiT v0.8.0-alpha on my Umbrel where I had LND v0.5.0-beta installed, and the LiT container fails to start due to this error: `2022-09-16 10:41:36.067 [ERR] LITD: Could not start subservers: error checking connected lnd version. at least version "v0.15.1, build tags 'signrpc,walletrpc,chainrpc,invoicesrpc'" is required`. A user wouldn't know what the issue is without digging into the container logs.

**Describe the solution you'd like**
Could we make it so this is error (and potentially others) are shown in the Web UI so that user can more easily debug their setup when something is misconfigured, a version mismatch, etc.

**Describe alternatives you've considered**
The only alternative atm is for a user to access the container logs to see this error.

### LINKED ISSUE #421: Show dependency issues in the UX
### Problem
As currently architected, at `litd` start up, in order to open the UX, all of the subservers need to startup meaning all of the `litd` dependencies (LND version, etc.) have to be aligned. With that behavior, for users using GUI experiences because of ease or lack of technical skills, when a `litd` update occurs that changes dependencies, they encounter an issue where the app or experience doesn't open due to dependency mismatch. However, given this user type, it can be difficult for them to debug the issue and understand the problem and solution. 

### Potential Solution
We could re-architect litd so that it can actually "startup" without all of the subservers being activated. With this architecture, we could show the user the relevant errors or dependency issues so that they can more easily debug. The example referenced in the attached issue would mean that we show the user in the UX that they need to be running `lnd 0.15.1` for litd to work properly. But, we've also run into issues in the past whereby the entire `litd` experience doesn't start because of a single subserver issue (Pool most of the time). With this architecture, we might be able to have everything else work, but in the Pool UX flag the error to the user to help them and us debug it. This would probably require an always-on RPC for the UI to obtain the current status or startup errors. 

An additional feature that could be a part of this is being able to configure the subservers from the UI. Then, there would be less flags that need to be specified via the CLI.

Referenced Issue: https://github.com/lightninglabs/lightning-terminal/issues/420

### PR DESCRIPTION
**Pull request by ellemouton**
This PR introduces a `Status` server to Lit. This server can be queried at any time to determine the 
status of any of the following servers: Lit, LND, Faraday, Loop, Pool. With this PR, Lit will also not shutdown 
unless stopped purposefully. The reason being so that if any parts of Litd fail to start, the status server will 
still be available for a front end to query to see what is up. 

Let's say for example, that Lit is started with Loop disabled (`loop-mode=disabled`) & then some error causes Pool
not to start up properly, then you can query the status server to see the following:
```
$ litcli status 

{
        "sub_servers": {
                "faraday": {
                        "disabled": false,
                        "running": true,
                        "error": ""
                },
                "lit": {
                        "disabled": false,
                        "running": true,
                        "error": ""
                },
                "lnd": {
                        "disabled": false,
                        "running": true,
                        "error": ""
                },
                "loop": {
                        "disabled": true,
                        "running": false,
                        "error": ""
                },
                "pool": {
                        "disabled": false,
                        "running": false,
                        "error": "pool not working"
                },
                "taproot-assets": {
                        "disabled": false,
                        "running": true,
                        "error": ""
                }
        }
}


```

Depends on: #616 

Replaces https://github.com/lightninglabs/lightning-terminal/pull/442

Fixes #421 
Fixes #420
Fixes #290

### CODE REVIEW

**Review by itsrachelfish — CHANGES_REQUESTED** [review]
I was able to get this integrated into terminal but while testing I noticed that when a subserver is disabled it completely disappears from the status list. We might want to change this because it means we won't be able to use the status server for feature detection.

For example when taproot assets are added to litd, it could be useful to query the status server to see if tapd is available. Currently there is no way to differentiate between "litd hasn't been upgraded, so tapd does not exist" and "litd has been upgraded, and tapd exists, but it is disabled"

**Comment by ellemouton** [comment]
thanks for the suggestion @itsrachelfish. Updated accordingly

File: litrpc/lit-status.proto
Comment (guggero) [inline litrpc/lit-status.proto:1]: nit: empty line at file start.

File: status/manager.go
Comment (guggero) [inline status/manager.go:46]: nit: should use `make()` instead if we're not pre-assigning values.

File: status/manager.go
Comment (guggero) [inline status/manager.go:36]: I'm probably missing something, but why can't the `subservers.Manager` keep track of disabled/running/error and we just have a simple RPC server struct that gets an instance of said manager?

File: subservers/manager.go
Comment (guggero) [inline subservers/manager.go:58]: I guess this is another hint that we might want to merge the two managers instead.

File: itest/litd_mode_integrated_test.go
Comment (guggero) [inline itest/litd_mode_integrated_test.go:444]: Shouldn't this have exactly the opposite meaning as in the proxy server should never be disabled? So we should never get the `disabledErr` returned?

File: itest/litd_mode_integrated_test.go
Comment (guggero) [inline itest/litd_mode_integrated_test.go:849]: Should we assert the success content here before returning?

**Review by guggero — COMMENTED** [review]
Awesome PR, great work!
I did a first pass to load all the context. I think the main thing we might want to change is merging of the two subserver managers (see inline comments).
I'll do a round of manual testing in the next round.

File: terminal.go
Comment (ViktorTigerstrom) [inline terminal.go:1142]: I think we should also set the `returnErr` here if `g.rpcProxy.Stop()` or `g.httpServer.Close()` fails?

File: terminal.go
Comment (ViktorTigerstrom) [inline terminal.go:253]: nit: could use a comment for this code block, given that the all the other code blocks in this function are preceded by comments.

File: status/manager.go
Comment (ViktorTigerstrom) [inline status/manager.go:112]: nit:
```suggestion
// SetEnabled marks the sub-server with the given name as enabled.
// This will silently fail if the referenced sub-server has not yet been registered.
func (s *Manager) SetEnabled(name string) {
```

File: status/manager.go
Comment (ViktorTigerstrom) [inline status/manager.go:33]: nit:
```suggestion
// Manager manages the status of any sub-server registered to it. It is also an
```

File: status/log.go
Comment (ViktorTigerstrom) [inline status/log.go:10]: nit:
```suggestion
// log is a logger that is initialized with no output filters. This
```

**Review by ViktorTigerstrom — COMMENTED** [review]
Very nice PR 🚀 🔥!!! My only comments are for minor things, so really great work! I've tested the functionality manually as well, and will also conduct thorough manual testing next review round 🚀!

File: status/manager.go
Comment (ellemouton) [reply status/manager.go:36]: the main reason is cause `subservers.Manager` is just meant to handle the external daemons and does not handle Lit and LND (both of which are kind of in their own category) .So seemed a bit contrived to try squash those in to the `subservers.Manager` since they dont fit the `SubServer` interface.

File: subservers/manager.go
Comment (ellemouton) [reply subservers/manager.go:58]: i think every subserver should have a status but not each thing that has a status is a subserver

File: status/manager.go
Comment (guggero) [reply status/manager.go:36]: Ah, okay, I didn't consider that. Okay, makes sense to keep them separate then.

Comment (ellemouton) [reply status/manager.go:36]: lemme know what you think :)

File: itest/litd_mode_integrated_test.go
Comment (ellemouton) [reply itest/litd_mode_integrated_test.go:855]: jip - good catch 👍  also updated the `runGRPCWebAuthTest` test to properly check success content

**Review by ellemouton — COMMENTED** [review]
Thanks @guggero  & @ViktorTigerstrom  🚀  updated!

File: terminal.go
Comment (guggero) [inline terminal.go:464]: We also need to set `lnd` as errored on line 444. I tested this in integrated mode where `lnd` couldn't start because of an already occupied listener port. The status shows `running: false` but `error: ""`.

Comment (guggero) [reply terminal.go:464]: And then, when trying to shut down (Ctrl+C), it just logs `LTND: Gracefully shutting down.` but doesn't quit... This could be hard to pin down though, so low prio/non-blocking.

File: terminal.go
Comment (guggero) [inline terminal.go:461]: I think we need to react to the `unlockChan` here somehow. I tested integrated mode where `lnd` was locked, and in that state it shows `running: false` for `lnd`. And I can't run `lncli --rpcserver localhost:8443 state` because I get `rpc error: code = Unknown desc = lnd is not running:`. Can we maybe count `lnd` as being locked as a successful startup and let `RPC` calls through? Otherwise we also won't be able to create a wallet or unlock using the 8443 port.

File: cmd/litcli/status.go
Comment (guggero) [inline cmd/litcli/status.go:28]: I wonder if we also should try calling `lnd`'s `State.GetState` RPC here and print that as well? Just as a nice addition for CLI users.

File: itest/litd_mode_integrated_test.go
Comment (guggero) [reply itest/litd_mode_integrated_test.go:450]: Ah okay, makes sense, thanks.

File: terminal.go
Comment (guggero) [reply terminal.go:461]: Also, again same issue with not being able to shut down while lnd is not unlocked yet. Maybe something small after all?

**Review by guggero — COMMENTED** [review]
Code looks great! Nothing to add there.

Did a round of manual tests and found some issues around lnd startup and unlock behavior.

File: terminal.go
Comment (ellemouton) [reply terminal.go:464]: for both of these, the issue is pre-existing. 
Currently (master branch) if you already have LND running & then try starting LiT in integrated mode, things hang on the `lnd.Main(...)` line. If you then shutdown (Ctrl+C) it will also just log `LTND: Gracefully shutting down.` and will log `LTND: Shutdown complete` about 10 seconds later but still the `lnd.Main(...)` method wont return and so Litd doesnt exit.  

I think the issue is that things block here: https://github.com/lightninglabs/lightning-terminal/blob/03f4bf99a9fd0313a8f219492dcdf25a88817169/terminal.go#L547

I'll open a separate PR to fix this. 

The one open question is: if the port is occupied, `lnd.Main` does not return. It hangs and retries. So there really isnt any way for LiT to set the `error:` in the status. So I think for now, just showing "running: false" is good enough since that should indicate to the user that something is wrong.

File: terminal.go
Comment (ellemouton) [reply terminal.go:467]: good catch! I have moved  the `g.statusMgr.SetRunning(subservers.LND)` line a bit earlier 👍

Comment (ellemouton) [reply terminal.go:467]: this also pointed to me to another pre-existing issue which is that LND's whitelisted calls are currently still checked for macaroons if they go through 8443. So i'll open another PR to fix that too :)

Comment (ellemouton) [reply terminal.go:467]: ok - added this PR: https://github.com/lightninglabs/lightning-terminal/pull/617

state requests should now work even while LND is still locked 👍

**Comment by ellemouton** [comment]
@AndySchroder  - I'd say this PR is pretty close. But just note that we wont merge this PR until the front end ready as it needs to be able to handle the case where some sub-servers are disabled. So if you are using the front-end then maybe wait. But if not, then I think you can try this PR. There are no DB changes or anything so if anything goes wrong (pls let us know!) then you can just revert to the last released version. 

As with any un-merged & un-released PR: please proceed at your own risk

**Review by itsrachelfish — APPROVED** [review]
tACK, the latest changes including the `disabled` flag are working with the front-end #620

**Review by ViktorTigerstrom — COMMENTED** [review]
Looks really good to me!
After some further manual testing, I've noticed some strange behaviour for subservers which are run in remote mode though.
1.
If running LiT with a subserver configured as in `remote` mode (specifically pool, loop & faraday), and if I then manually shutdown the remote subserver, the `lncli status` command will still mark the specific subserver as "running".
2.
Even more strangely, for some reason which I can't figure out why when looking at the code, if I have shutdown a subserver and then start LiT with a subserver configured as in `remote` mode (specifically tried with pool & loop), the `lncli status` command will still mark the specific subserver as "running".
This is despite the LiT logs clearing showing that we can't connect to the subserver. See the log snippet below:

```
2023-09-12 02:11:35 ----------------------------------------------------------
2023-09-12 02:11:35  Lightning Terminal (LiT) by Lightning Labs               
2023-09-12 02:11:35                                                           
2023-09-12 02:11:35  LND Operating mode      remote                               
2023-09-12 02:11:35  LND Node status         online                               
2023-09-12 02:11:35  LND Alias               dave                               
2023-09-12 02:11:35  LND Version             0.16.4-beta commit=v0.16.4-beta                               
2023-09-12 02:11:35  LiT Version             0.10.5-alpha commit=v0.10.5-alpha-68-g5c8986823ae8483e21c5b01b3c7ef7bc9c233282                               
2023-09-12 02:11:35  Web interface           0.0.0.0:8443 (open https://localhost:8443 in your browser)                                    
2023-09-12 02:11:35 ----------------------------------------------------------
2023-09-12 02:11:35 2023-09-12 00:11:35.436 [INF] LITD: Connecting basic lnd client
2023-09-12 02:11:35 2023-09-12 00:11:35.638 [INF] LITD: Connecting full lnd client
2023-09-12 02:11:35 2023-09-12 00:11:35.638 [INF] LNDC: Creating lnd connection to dave:10009
2023-09-12 02:11:35 2023-09-12 00:11:35.638 [INF] LNDC: Connected to lnd
2023-09-12 02:11:35 2023-09-12 00:11:35.640 [INF] LNDC: Waiting for lnd to unlock
2023-09-12 02:11:35 2023-09-12 00:11:35.641 [INF] LNDC: Wallet state of lnd is now: Lnd main server is ready for requests
2023-09-12 02:11:35 2023-09-12 00:11:35.649 [INF] LNDC: lnd version: v0.16.4-beta, build tags 'signrpc,walletrpc,chainrpc,invoicesrpc,peersrpc'
2023-09-12 02:11:35 2023-09-12 00:11:35.649 [INF] LNDC: Using network regtest
2023-09-12 02:11:35 2023-09-12 00:11:35.649 [INF] LNDC: Waiting for lnd to be fully synced to its chain backend, this might take a while
2023-09-12 02:11:35 2023-09-12 00:11:35.659 [INF] LNDC: lnd is now fully synced to its chain backend
2023-09-12 02:11:35 2023-09-12 00:11:35.660 [INF] SSVR: Opening sqlite3 database at: /root/.tapd/data/regtest/tapd.db
2023-09-12 02:11:35 2023-09-12 00:11:35.663 [INF] SRVR: Version: 0.2.3-alpha commit=v0.2.3, build=production, logging=default, debuglevel=info
2023-09-12 02:11:35 2023-09-12 00:11:35.663 [INF] SRVR: Active network: regtest
2023-09-12 02:11:35 2023-09-12 00:11:35.663 [INF] RPCS: Validating RPC requests based on macaroon at: /root/.tapd/data/regtest/admin.macaroon
2023-09-12 02:11:35 2023-09-12 00:11:35.664 [INF] GRDN: Starting ChainPlanter
2023-09-12 02:11:35 2023-09-12 00:11:35.664 [INF] GRDN: Retrieved 0 non-finalized batches from DB
2023-09-12 02:11:35 2023-09-12 00:11:35.665 [INF] GRDN: Starting asset custodian
2023-09-12 02:11:35 2023-09-12 00:11:35.665 [INF] GRDN: Gardener for ChainPlanter now active!
2023-09-12 02:11:35 2023-09-12 00:11:35.665 [INF] GRDN: Loading pending inbound asset events
2023-09-12 02:11:35 2023-09-12 00:11:35.666 [INF] FRTR: Starting ChainPorter
2023-09-12 02:11:35 2023-09-12 00:11:35.666 [INF] UNIV: Starting FederationEnvoy
2023-09-12 02:11:35 2023-09-12 00:11:35.666 [INF] UNIV: Adding new Universe server to Federation, addrs=([]universe.ServerAddr) {
2023-09-12 02:11:35 }
2023-09-12 02:11:35 
2023-09-12 02:11:35 2023-09-12 00:11:35.667 [INF] GRDN: Resuming 0 pending inbound asset events
2023-09-12 02:11:35 2023-09-12 00:11:35.667 [INF] GRDN: Loading wallet transactions starting at block height 0
2023-09-12 02:11:35 2023-09-12 00:11:35.667 [WRN] UNIV: unable to add universe servers: <nil>
2023-09-12 02:11:35 2023-09-12 00:11:35.667 [INF] RPCS: Starting RPC Server
2023-09-12 02:11:35 2023-09-12 00:11:35.667 [INF] LITD: Starting LiT macaroon service
2023-09-12 02:11:35 2023-09-12 00:11:35.668 [INF] GRDN: Checking 4 wallet transactions for inbound assets, this might take a while
2023-09-12 02:11:35 2023-09-12 00:11:35.669 [INF] GRDN: Starting main custodian event loop
2023-09-12 02:11:35 2023-09-12 00:11:35.720 [INF] AUTO: Starting Autopilot Client
2023-09-12 02:11:35 2023-09-12 00:11:35.726 [INF] LITD: Starting LiT session server
2023-09-12 02:11:35 2023-09-12 00:11:35.726 [INF] LITD: Starting LiT account service
2023-09-12 02:11:35 2023-09-12 00:11:35.727 [INF] LITD: Starting LiT middleware manager
2023-09-12 02:11:35 2023-09-12 00:11:35.728 [INF] LITD: Internal sub server startup complete
2023-09-12 02:11:36 2023-09-12 00:11:36.458 [WRN] GRPC: [core] grpc: addrConn.createTransport failed to connect to {davepool:12010 davepool:12010 <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp: lookup davepool on 127.0.0.11:53: no such host". Reconnecting...
2023-09-12 02:11:38 2023-09-12 00:11:38.208 [WRN] GRPC: [core] grpc: addrConn.createTransport failed to connect to {davepool:12010 davepool:12010 <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp: lookup davepool on 127.0.0.11:53: no such host". Reconnecting...
2023-09-12 02:11:38 2023-09-12 00:11:38.292 [INF] LITD: Handling gRPC request: /litrpc.Status/SubServerStatus
2023-09-12 02:11:38 2023-09-12 00:11:38.293 [INF] LITD: Handling gRPC request: /lnrpc.State/GetState
2023-09-12 02:11:41 2023-09-12 00:11:41.214 [WRN] GRPC: [core] grpc: addrConn.createTransport failed to connect to {davepool:12010 davepool:12010 <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp: lookup davepool on 127.0.0.11:53: no such host". Reconnecting...
```
The `lncli status` command executed in that log snippet marked `pool` as `"running":true` for some reason.

Other than this, this PR looks great :rocket::fire:!!

**Comment by ellemouton** [comment]
> If running LiT with a subserver configured as in remote mode (specifically pool, loop & faraday), and if I then manually shutdown the remote subserver, the lncli status command will still mark the specific subserver as "running".

Yep - this is expected. This initial version is only useful at startup time and it does not ping the subservers or update the status during the lifetime of the daemon. This is something that can be added in future

**Comment by ellemouton** [comment]
> Even more strangely, for some reason which I can't figure out wh

cool - that sounds wrong :+1: I'll look into that 👍

File: subservers/manager.go
Comment (ellemouton) [reply subservers/manager.go:69]: At the moment, this is is only being called when LiT itself is also shutting down. meaning the status server anyways wont be up or meaningful. 

I think in a future PR we can address restarts of sub-servers etc

File: subservers/manager.go
Comment (ellemouton) [reply subservers/manager.go:113]: yep - this PR currently only handles start up errors.

**Comment by ellemouton** [comment]
> (2). Even more strangely, for some reason which I can't figure out why when looking a

Ok so this is cause currently for remote sub-servers, we dont block on the connection being successful. Instead we give it:

```
grpc.WithConnectParams(grpc.ConnectParams{
	Backoff:           backoff.DefaultConfig,
	MinConnectTimeout: defaultConnectTimeout,
})
```

and return the unconnected connection so that LiT continues to maintain the connection if the remote server gets restarted or something. I dont think we want to change this behaviour right now. And I think in most cases, the pool/faraday/loop  etc servers are run in integrated mode in anycase. 

I think in a follow up PR, we can add a heartbeat mechanism to the status server to give more live updates regarding sub-server statuses.

**Review by ViktorTigerstrom — APPROVED** [review]
tACK LGTM! Great work :fire: :rocket: :fire:

File: subservers/manager.go
Comment (ViktorTigerstrom) [reply subservers/manager.go:69]: Ok makes sense :)!

**Comment by ViktorTigerstrom** [comment]
> > (2). Even more strangely, for some reason which I can't figure out why when looking a
> 
> Ok so this is cause currently for remote sub-servers, we dont block on the connection being successful. Instead we give it:
> 
> ```
> grpc.WithConnectParams(grpc.ConnectParams{
> 	Backoff:           backoff.DefaultConfig,
> 	MinConnectTimeout: defaultConnectTimeout,
> })
> ```
> 
> and return the unconnected connection so that LiT continues to maintain the connection if the remote server gets restarted or something. I dont think we want to change this behaviour right now. And I think in most cases, the pool/faraday/loop etc servers are run in integrated mode in anycase.
> 
> I think in a follow up PR, we can add a heartbeat mechanism to the status server to give more live updates regarding sub-server statuses.

Ok makes sense, thanks for checking! Sound like a good plan

File: config.go
Comment (ViktorTigerstrom) [inline config.go:196]: nit:
```suggestion
	PoolMode string       `long:"pool-mode" description:"The mode to run pool in, either 'integrated' (default), 'remote' or 'disable'. 'integrated' means poold is started alongside the UI and everything is stored in pool's main data directory, configure everything by using the --pool.* flags. 'remote' means the UI connects to an existing poold node and acts as a proxy for gRPC calls to it. 'disable' means that LiT is started without pool." choice:"integrated" choice:"remote" choice:"disable"`
```

**Review by guggero — APPROVED** [review]
Amazing work. tACK, LGTM :tada:

File: rpc_proxy.go
Comment (ellemouton) [reply rpc_proxy.go]: done :)

File: status/manager.go
Comment (ellemouton) [reply status/manager.go:115]: done :)

**Comment by guggero** [comment]
I assume we'll want to wait for https://github.com/lightninglabs/lightning-terminal/pull/620 before merging this?

**Comment by ellemouton** [comment]
@guggero  - yes :)

**Pull request merged**
